rfcs/031-monorepo.md

```diff
@@ -0,0 +1,100 @@
+- Feature Name: monorepo
+- Start Date: 2021-03-30
+- RFC PR: [amundsen-io/rfcs#0030](https://github.com/amundsen-io/rfcs/pull/30)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Move to Monorepo
+
+## Summary
+
+At the moment, Amnundsen project is divided into 6 repositories: 
+	- amundsen
+	- amundsenfrontendlibrary
+	- amundsenmetadatalibrary
+	- amundsensearchlibrary
+	- amundsencommon 
+	- amudnsendatabuilder
+
+This RFC proposes merging all of these repositories into a single repo.
+
+There will be no change to the built artifacts. Users who use PyPi packages or Docker images will have no change to their deployments. Users with forks or submodule based customization will have to make minor changes to their CI/CD pipelines to pull from the correct directories, but the required changes will be minor.
+
+
+## Motivation
+
+The submodule architecture creates the following issues and is making it hard for companies to adopt Amundsen, and to customize it accordingly. It also causes substantial developer toil in the form of having to create multiple small PRs to merge one 
+
+1. Dependency Management: Python packages are pretty much the same for each proxy, which makes it hard to sync across all the above repositories. As a result most of the time, each proxy is running its own version of the dependency. 
```

Comment:
Another approach (or used alonside changes from this RFC to strengthen the dependency management even more) would be introducing the pip constraints files https://pip.pypa.io/en/stable/user_guide/#constraints-files

Reply:
these are new to me! That's super cool, added.